device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/edge/DirectMethodResponse.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Copyright (c) Microsoft. All rights reserved.
+ *  Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+package com.microsoft.azure.sdk.iot.device.edge;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+public class DirectMethodResponse
+{
+    private static final String STATUS_KEY_NAME = "status";
+    @Expose(serialize = false)
+    @SerializedName(STATUS_KEY_NAME)
+    @Getter
+    private int status;
+
+    private static final String PAYLOAD_KEY_NAME = "payload";
+    @Expose(serialize = false)
+    @SerializedName(PAYLOAD_KEY_NAME)
+    private JsonElement payload;
+
+    //empty constructor for gson
+    private DirectMethodResponse()
+    {
+
+    }
+
+    public DirectMethodResponse(String json)
+    {
+        DirectMethodResponse result = new GsonBuilder().create().fromJson(json, DirectMethodResponse.class);
+
+        this.payload = result.payload;
+        this.status = result.status;
+    }
+
+    /**
+     * Return the DirectMethodResponse payload in JsonElement type
+     * @return the DirectMethodResponse payload in JsonElement type
+     */
+    public JsonElement getPayloadAsJsonElement()
+    {
+        return payload;
+    }
+
+    /**
+     * Return the DirectMethodResponse payload in String type
+     * @return the DirectMethodResponse payload in String type
+     */
+    public String getPayloadAsString()
+    {
+        return getPayloadAsCustomType(String.class);
+    }
+
+    /**
+     * Return the DirectMethodResponse payload in a custom type of your choosing
+     * Use this if you wish to deserialize to a specific type using a deserialization library of your choice
```

Comment:
Nope, this statement belongs on the `getPayloadAsString()` method. This one uses gson to deserialize.